source/types/ut_assert_result_base.tps

```diff
@@ -0,0 +1,13 @@
+create or replace type ut_assert_result_base as object
+(
+  result   integer(1),
+  message  varchar2(4000 char),
+  expected varchar2(4000 char),
+  actual   varchar2(4000 char),
+  end_time timestamp with time zone,
+  not instantiable member procedure add_element( self in out nocopy ut_assert_result_base, an_assertion ut_assert_result_base ),
```

Comment:
You've raised an interesting point.
There might a lot of different assertions that give different outcomes.
Most will compare expected vs actual but some will not (probably).
I'll have a look into RSpec code and try to figure out how it's done there - they have lots of different assertions.
The trouble is: how to separate the assertions from reporting of the assert outcomes?
I'd like to give the reporter full control on how to report on asserts.
If we do not provide consistent assertion results to be interpreted by reporter then the reporting will sneak int the assertions - which is the case in the code prior to my PR.
Something to think about.
Maybe we can stick for now to expected+actual as i have no better idea on how to give full control on formatting to the reporter otherwise.
The only thing i'll do is that i'll move those to the ut_assert_result.
As you've noticed the base class in the proposed form and usage doesn't make much sense.